docker/docker_client.go

```diff
@@ -533,6 +533,8 @@ func (c *dockerClient) getBearerToken(ctx context.Context, challenge challenge,
 	defer res.Body.Close()
 	switch res.StatusCode {
 	case http.StatusUnauthorized:
+		err := client.HandleErrorResponse(res)
+		logrus.Debugf("Server response when trying to obtain an access token: \n%q", err.Error())
 		return nil, ErrUnauthorizedForCredentials
```

Comment:
what does break API mean?
```
return nil, errors.Wrapf(err, "Server response when trying to obtain an access token")
```

Reply:
In current version Buildah and Podman can just look at the return code and compare it to ErrUnauthorizedForCredentials
If we wrap the error with the message, then podman and buildah will no longer get ErrUnauthorizedForCredentials back, they would have to take the error returned and change it to 
if errors.Cause(err) == ErrUnauthorizedForCredentials

Any time the users of a library have to change their code, then you have broken ABI, if the callers do not have to change their code, then ABI remained the same. 

https://en.wikipedia.org/wiki/Application_binary_interface

Reply:
Ok, Thanks, PTAL https://github.com/containers/image/pull/719